lib/gooddata/helpers/global_helpers_params.rb

```diff
@@ -52,9 +52,12 @@ def encode_hidden_params(params)
         encode_params(params, ENCODED_HIDDEN_PARAMS_KEY)
       end
 
-      # Decodes params as they came from the platform
+      # Decodes params as they came from the platform. It also replace reference parameters ${} in gd_encoded_params (if any) by actual value
       # The "data" key is supposed to be json and it's parsed - if this
-      def decode_params(params)
+      # @params Parameter hash need to be decoded
+      # @option options [Boolean] :resolve_reference_params (true) Resolve reference parameters in gd_encoded_params or not
+      # @return [Hash] Decoded parameters
+      def decode_params(params, options = { :resolve_reference_params => true })
```

Comment:
Actually I'm not sure about this. It breaks the default behaviour of `decode_params`. IMOP,  `decode_params` does not support `resolve_reference_params` by default and we need to enable this feature at https://github.com/gooddata/gooddata-ruby/blob/develop/lib/gooddata/bricks/middleware/decode_params_middleware.rb#L16 and the same place in `gooddata-bricks`. Please feedback @korczis, @zsvoboda